### PR TITLE
Addressed Security Concerns

### PR DESCRIPTION
**Pull request by kave**
### The purpose of this is to address Security concerns brought up by security team

- addressed security concerns about query data validation in calendar filter pdf generation, and added an option to filters to allow post requests
- fixed url routing for rendering directory cordrays pdf
- explicitly stated jinja2 to autoescape in templates

## Additions
- new setting variables
- new view for Directory Cordray PDF
- added option to allow POST requests in filter macro

## Changes

- Calendar filter download is now a post request wrapped in a form and also validated on the backend

## Review

- @rosskarchner 
- @kurtw 

## Checklist

* [] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [ ] New functions include new tests
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices 
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: cfgov/v1/forms.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from django import forms
+
+
+class CalenderPDFFilterForm(forms.Form):
+    filter_calendar = forms.CharField()
+    filter_range_date_gte = forms.DateField(input_formats=['%Y-%m-%d'])
+    filter_range_date_lte = forms.DateField(input_formats=['%Y-%m-%d'])
+
+    def clean_filter_calendar(self):
+        return self.cleaned_data['filter_calendar'].replace(' ', '+')
```
Comment (kave) [inline cfgov/v1/forms.py]: spaces had to be replaced by '+' for the filtering to work as expected

File: cfgov/core/services.py [hunk 2]
```diff
@@ -68,8 +72,18 @@ def generate_pdf(self):
                 self.get_filename())
             return response
 
-    def get(self, *args, **kwargs):
-        return self.generate_pdf()
+    def post(self, request):
+        form = CalenderPDFFilterForm(request.POST)
```
Comment (kave) [inline cfgov/core/services.py]: wrapped the request in a form that validates if ranges are actually dates else it just throws an exception.

File: cfgov/cfgov/settings/base.py [hunk 3]
```diff
@@ -134,7 +135,6 @@
 
 # Sheer related settings
 
-SHEER_SITES = [Path(os.path.dirname(__file__)).ancestor(2) + '/v1/jinja2/v1',
-               Path(__file__).ancestor(4).child('docs'), STATIC_ROOT]
+SHEER_SITES = [V1_TEMPLATE_ROOT, Path(__file__).ancestor(4).child('docs'), STATIC_ROOT]
```
Comment (kurtw) [inline cfgov/cfgov/settings/base.py]: Should we just remove the 'docs' site? FEWD's said they don't use it.

File: cfgov/core/services.py (hunk 2 above)
Comment (kurtw) [inline cfgov/core/services.py]: And why change it to a post?

File: cfgov/v1/__init__.py [hunk 4]
```diff
@@ -11,6 +11,7 @@
 def environment(**options):
     options.setdefault('extensions', []).append(CompressorExtension)
     env = sheerlike_environment(**options)
+    env.autoescape = True
```
Comment (kurtw) [inline cfgov/v1/__init__.py]: What's up with this?

File: cfgov/core/services.py (hunk 2 above)
Comment (kave) [inline cfgov/core/services.py]: @kurtw so i could wrap the request in a form, it makes getting individual values simpler & validation cleaner

File: cfgov/v1/forms.py (hunk 1 above)
Comment (kurtw) [inline cfgov/v1/forms.py]: As long as the data is part of the query string, this should be fine.

File: cfgov/cfgov/settings/base.py (hunk 3 above)
Comment (kave) [inline cfgov/cfgov/settings/base.py]: hmmm we could, looked like it had some useful template convention related information. i guess it doesn't hurt to leave it in?

File: cfgov/core/services.py (hunk 2 above)
Comment (kurtw) [inline cfgov/core/services.py]: Makes sense :+1:

File: cfgov/cfgov/settings/base.py (hunk 3 above)
Comment (anselmbradford) [inline cfgov/cfgov/settings/base.py]: One part we do occasionally reference is `/docs/sheer-layouts/`, but yeah overall we don't use it. At some point we'd like to bring it back though, if that matters for this code.

File: cfgov/v1/__init__.py (hunk 4 above)
Comment (kave) [inline cfgov/v1/__init__.py]: one of the security findings was "jinja2 sets autoescape to False." So making sure the env explicitly sets this

File: cfgov/v1/forms.py (hunk 1 above)
Comment (kave) [inline cfgov/v1/forms.py]: :+1:

File: cfgov/v1/__init__.py (hunk 4 above)
Comment (kurtw) [inline cfgov/v1/__init__.py]: Okay so if it's autoescaping now then there's a ton of escaping done in the templates that is redundant code. Idk if it should be in the scope of this PR but it's something to be aware of.

File: cfgov/v1/jinja2/v1/_includes/macros/filters.html [hunk 5]
```diff
@@ -99,6 +99,7 @@
     'action':                   '/',
     'expand_label':             'Filter',
     'submit_label':             'Apply filters',
+    'form_method':              'get',
```
Comment (anselmbradford) [inline cfgov/v1/jinja2/v1/_includes/macros/filters.html]: Add entry for `user_options.form_method` to the `user_options` code comment docs [here](https://github.com/cfpb/cfgov-refresh/blob/security-fixes/cfgov/v1/jinja2/v1/_includes/macros/filters.html#L60)

File: cfgov/v1/__init__.py (hunk 4 above)
Comment (kave) [inline cfgov/v1/__init__.py]: hmm i'll make a note of this and look into it in another PR

File: cfgov/v1/jinja2/v1/_includes/macros/filters.html (hunk 5 above)
Comment (kave) [inline cfgov/v1/jinja2/v1/_includes/macros/filters.html]: aah yes, taken care off

**Pull request merged**